connectors/sources/github.py

```diff
@@ -2075,6 +2076,28 @@ async def _fetch_access_control(self, repo_name):
                         access_control.append(_prefix_email(email=user_email))
         return access_control
 
+    def _is_repo_processed(self, repo_name):
+        if self._checkpoint is None:
+            self._logger.debug(f"{repo_name} was not processed by this job before.")
+            return False
+
+        processed = self._checkpoint.get(repo_name)
+
+        if processed:
+            self._logger.info(
+                f"Skipping repository {repo_name} - has already been processed."
+            )
+
+        return processed
+
+    def _mark_repo_as_processed(self, repo_name):
+        self._logger.debug(f"Setting a checkpoint for {repo_name}.")
+        if self._checkpoint is None:
+            self._checkpoint = {}
+
+        when = datetime.now(timezone.utc)
+        self._checkpoint[repo_name] = when
```

Comment:
Mostly it's for debugging/tracing, there's no real need to have a timestamp here.

Reply:
I'm okay with the timestamp, I just thought if it's used for comparison at some point it might run into issues with timezones, but maybe I'm overthinking things anyway.